src/devices/Vl53L1X/Vl53L1X.cs

```diff
@@ -0,0 +1,1022 @@
+        }
+
+        /// <summary>
+        ///     This function programs the distance mode (1 = Short, 2 = Long).
+        ///     Short mode maximum distance is limited to 1.3m but results in a better ambient immunity.
+        ///     Long mode can range up to 4 m in the dark with a timing budget of200 ms.
+        /// </summary>
+        /// <param name="mode">The mode to program</param>
+        public void SetDistanceMode(DistanceMode mode)
+        {
+            TimingBudget timingBudget = GetTimingBudgetInMs();
+
+            switch (mode)
+            {
+                case DistanceMode.Short:
+                    WriteRegister((ushort)Registers.PHASECAL_CONFIG__TIMEOUT_MACROP, 0x14);
+                    WriteRegister((ushort)Registers.RANGE_CONFIG__VCSEL_PERIOD_A, 0x07);
+                    WriteRegister((ushort)Registers.RANGE_CONFIG__VCSEL_PERIOD_B, 0x05);
+                    WriteRegister((ushort)Registers.RANGE_CONFIG__VALID_PHASE_HIGH, 0x38);
+                    WriteUInt16((ushort)Registers.SD_CONFIG__WOI_SD0, 0x0705);
+                    WriteUInt16((ushort)Registers.SD_CONFIG__INITIAL_PHASE_SD0, 0x0606);
+                    break;
+                case DistanceMode.Long:
+                    WriteRegister((ushort)Registers.PHASECAL_CONFIG__TIMEOUT_MACROP, 0x0A);
+                    WriteRegister((ushort)Registers.RANGE_CONFIG__VCSEL_PERIOD_A, 0x0F);
+                    WriteRegister((ushort)Registers.RANGE_CONFIG__VCSEL_PERIOD_B, 0x0D);
+                    WriteRegister((ushort)Registers.RANGE_CONFIG__VALID_PHASE_HIGH, 0xB8);
+                    WriteUInt16((ushort)Registers.SD_CONFIG__WOI_SD0, 0x0F0D);
+                    WriteUInt16((ushort)Registers.SD_CONFIG__INITIAL_PHASE_SD0, 0x0E0E);
+                    break;
+            }
+
+            SetTimingBudgetInMs(timingBudget);
+        }
+
+        /// <summary>
+        ///     This function returns the current distance mode (1 = Short, 2 = Long).
+        /// </summary>
+        /// <returns>The currently used distance mode.</returns>
+        /// <exception cref="ArgumentOutOfRangeException">If reported distance mode is unknown</exception>
+        public DistanceMode GetDistanceMode()
+        {
+            byte temp = ReadByte((ushort)Registers.PHASECAL_CONFIG__TIMEOUT_MACROP);
+            if (temp != 0x14 && temp != 0xA)
+            {
+                return DistanceMode.Unknown;
+            }
+
+            return temp == 0x14 ? DistanceMode.Short : DistanceMode.Long;
+        }
+
+        /// <summary>
+        ///     This function programs the intermeasurement period in ms.
+        ///     Intermeasurement period must be >/= timing budget. This condition is not checked by the API,
+        ///     the customer has the duty to check the condition. Default = 100 ms.
+        /// </summary>
+        /// <param name="interMeasMs">The inermeasurement period to set.</param>
+        public void SetInterMeasurementInMs(ushort interMeasMs)
```

Comment:
this one should be perfect as a property as well

Reply:
I'm now using properties wherever possible.